### ISSUE TITLE
Net::SSH fails on Windows with Pageant - Net::SSH::Exception: Creation of file mapping failed with error: 998

### ISSUE THREAD

**Issue by CpuID** [net-ssh/net-ssh#180]
Fires the exception located in https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/authentication/pageant.rb - Line 341

Specifically relates to issue a user experienced here - https://github.com/test-kitchen/test-kitchen/issues/448

Reproducible on Windows 8.1 64bit, by just having pageant.exe running, with or without any keys added.

Exception:

```
#<Net::SSH::Exception: Creation of file mapping failed with error: 998>
```

Exiting the pageant.exe process allows connections to proceed.

**Comment by chiefy** [comment]
@chrahunt I can't even recall if I issued a bug report w/ Ruby - I know I meant to.

**Comment by jamgregory** [comment]
Hey all. I've just come across this issue myself while trying to use Capistrano in Ruby 2.0 on Windows 7 x64. 

I'm afraid that I don't have much knowledge of the Windows SDK myself, but I did come across this post while searching for similar 998 errors:
http://www.tech-archive.net/Archive/Development/microsoft.public.win32.programmer.kernel/2009-08/msg00128.html

Not sure if that sheds any light on why this is happening, or what might need changing/fixing in Net::SSH to make this work.

**Comment by pleaseking** [comment]
This should be fixed by this pull request: https://github.com/net-ssh/net-ssh/pull/239

**Comment by sschuberth** [comment]
@CpuID Looks this issue is fixed, so would you mind closing it?

**Comment by thargor** [comment]
This issue is not fixed by #239
